runtime/service.go

```diff
@@ -415,6 +416,33 @@ func newServer() (*ttrpc.Server, error) {
 	return ttrpc.NewServer(ttrpc.WithServerHandshaker(ttrpc.UnixSocketRequireSameUser()))
 }
 
+func dialVsock(ctx context.Context, contextID uint32, port uint32) (net.Conn, error) {
+	// VM should start within 200ms, vsock dial will make retries at 100ms, 200ms, 400ms, 800ms and 1.6s
+	const (
+		retryCount      = 5
+		initialDelay    = 100 * time.Millisecond
+		delayMultiplier = 2
+	)
+
+	var lastErr error
+	var currentDelay = initialDelay
+	for i := 1; i <= retryCount; i++ {
+		conn, err := vsock.Dial(contextID, port)
+		if err == nil {
+			return conn, nil
+		}
+
+		log.G(ctx).WithError(err).Warnf("vsock dial failed (attempt %d of %d), will retry in %s", i, retryCount, currentDelay)
```

Comment:
I think this warning will end up firing more than we'd like to see in normal operations. Let's leave it in place for now, but we will probably want to change the severity and/or wording once we've seen this in operation a while. My hunch is that the 400ms attempt is the one that'll hit most frequently.